mapie/utils.py

```diff
@@ -5,21 +5,26 @@
 import numpy as np
 from sklearn.base import ClassifierMixin, RegressorMixin
 from sklearn.model_selection import BaseCrossValidator, KFold, LeaveOneOut
-from sklearn.utils.validation import _check_sample_weight, _num_features
+from sklearn.utils.validation import (
+    _check_sample_weight,
+    _num_features
+)
 from sklearn.utils import _safe_indexing
 
-from ._typing import ArrayLike
+from ._typing import ArrayLike, NDArray
 
 
 def check_null_weight(
-    sample_weight: ArrayLike, X: ArrayLike, y: ArrayLike
-) -> Tuple[ArrayLike, ArrayLike, ArrayLike]:
+    sample_weight: Optional[ArrayLike],
```

Comment:
Shouldn't an optional argument be placed after mandatory ones ?

Reply:
Yes, but the subtlety here is that the type `Optional[ArrayLike]` is nothing but a synonym of `Union[ArrayLike, None]`, so the type is valid, and python knows that it is not optional since there is no default value.